def convert_to_int(value: str):
   try:
       result = int(value)
       print(f"преобразование строки в целое число - успешно: {result}" )
   except ValueError:
       print(f"нельзя преобразовать строку в число")
   except Exception as ex:
       print(f"нельзя преобразовать строку в число {ex}")       
   finally:
       print("попытка преобразования завершена")

print("Тест 1:"), convert_to_int("123")
print("Тест 2:"), convert_to_int("ппп")
print("Тест 3:"), convert_to_int([1, 2, 3])

# Упражнение 1: Написать функцию с обработкой ошибок для преобразования строки в число

# Задание: Создать функцию convert_to_int(value), которая пытается преобразовать строку в целое число. Обработать случаи, когда преобразование невозможно, и когда возникает любое другое исключение.

# Пошаговая инструкция:
# •	Создайте функцию convert_to_int(value).
# •	Используйте блок try для попытки преобразования строки в целое число с помощью функции int().
# •	Обработайте исключение ValueError, выведя сообщение о невозможности преобразования.
# •	Обработайте любое другое исключение, выведя сообщение о неожиданной ошибке.
# •	Добавьте блок finally, который выведет сообщение о завершении попытки преобразования.

# Проверка:
# •	Протестируйте функцию с корректной строкой, например, "123".
# •	Протестируйте функцию с некорректной строкой, например, "abc".
# •	Протестируйте функцию с другим типом данных, например, списком [1, 2, 3].